### ISSUE TITLE
pidfile should be world-readable

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3024]
|<img alt="mthaddon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| mthaddon reported|
|-|-|
|Trac ID|trac#3024|
|Type|defect|
|Created|2008-02-07 17:55:52Z|

The pidfile for a twisted daemon should be world-readable. This allows the process to be monitored.

<details><summary>Searchable metadata</summary>

```
trac-id__3024 3024
type__defect defect
reporter__mthaddon mthaddon
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1202406952000000 1202406952000000
changetime__1217110893000000 1217110893000000
version__None None
owner__ 
cc__exarkun cc__spiv cc__zectbumo
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This has a lot of overlap with [#10647](https://github.com/twisted/twisted/issues/10647).  Maybe it's even a duplicate.  The pidfile is created with whatever permissions the umask dictates.  [#10647](https://github.com/twisted/twisted/issues/10647) lets you change the umask `twistd` sets.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [24393]) Merge twistd-umask-966

Author: exarkun
Reviewer: washort
Fixes: [#10647](https://github.com/twisted/twisted/issues/10647)
Fixes: [#3024](https://github.com/twisted/twisted/issues/3024)

Add a `--umask` option to `twistd` to allow the file creation mask
it sets to be customized.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
